docs/software_requirements/timers.sdoc

```diff
@@ -26,12 +23,334 @@ UID: ZEP-SRS-4-2
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-12
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Start timer counting values
+STATEMENT: >>>
+When a timer is started with the start mechanism, the timer starts counting with the given persiod and duration values
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-13
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer stop
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to stop a running timer prematurely.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-14
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer stop invocation
+STATEMENT: >>>
+When a Timer is stopped with the provide stop mechanism, the stop function is called if one is defined.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-15
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to read the status of a timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-16
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-17
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer status read reset
+STATEMENT: >>>
+When the timer is read via the timer status read mechanism, the status shall be reset.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-18
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer thread synchronization
+STATEMENT: >>>
+The Zephyr RTOS shall provide a mechanism to synchronize a thread with a defined and initialized timer.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-19
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer expire when thread synchronization is used
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the thread shall be blocked until the timer expires.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-20
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer is stopped when thread synchronization is used
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the thread shall be blocked until the timer is stopped.
+<<<
+RELATIONS:
+- TYPE: Parent
+  VALUE: ZEP-SYRS-18
+
+[REQUIREMENT]
+UID: ZEP-SRS-4-21
+STATUS: Draft
+TYPE: Functional
+COMPONENT: Timers
+TITLE: Timer thread synchronization status reset
+STATEMENT: >>>
+When the thread synchronization mechanism is used, the status of the timer shall be reset.
```

Comment:
Perhaps it's different views on semantics, I'd expect that syncing a thread on a timer should not affect the timer itself, since otherwise the operation will behave differently for the first and the following threads trying to sync at the same time. The current implementation in https://github.com/zephyrproject-rtos/zephyr/blob/911abc33e68b81c0490fa4b32a93d7d726640b73/kernel/timer.c#L295 resets the timer's status field after the timer expired and the first thread waiting for the timer runs again over this line. From the timer's perspective this happens at an arbitrary point in time as the (now running) thread might have spend an unspecified amount of time in the ready queue. 

Not saying this is false per se, just that I am not really getting it yet, why things are the way they are.